src/main/java/com/googlecode/download/maven/plugin/internal/WGet.java

```diff
@@ -332,50 +371,81 @@ private void unpack(File outputFile) throws NoSuchArchiverException {
+
         } else if (StringUtils.isNotBlank(serverId)) {
             getLog().debug("providing custom authentication for " + serverId);
             Server server = settings.getServer(serverId);
-            if (server == null)
+            if (server == null) {
                 throw new MojoExecutionException(String.format("Server %s not found", serverId));
+            }
             getLog().debug(String.format("serverId %s supplies username: %s and password: ***",  serverId, server.getUsername() ));
-            authenticationInfo.setUserName(server.getUsername());
-            authenticationInfo.setPassword(decrypt(server.getPassword(), serverId));
-        }
 
-        ProxyInfo proxyInfo = this.wagonManager.getProxy(repository.getProtocol());
+            credentialsProvider.setCredentials(
+                    new AuthScope(this.uri.getHost(), this.uri.getPort()),
+                    new UsernamePasswordCredentials(server.getUsername(), decrypt(server.getPassword(), serverId)));
+
+        }
 
-        wagon.connect(repository, authenticationInfo, proxyInfo);
-        wagon.get(file, outputFile);
-        wagon.disconnect();
-        if (downloadMonitor != null) {
-            wagon.removeTransferListener(downloadMonitor);
+        final HttpRoutePlanner routePlanner;
+        ProxyInfo proxyInfo = this.wagonManager.getProxy(this.uri.getScheme());
+        if (proxyInfo != null && proxyInfo.getHost() != null && ProxyInfo.PROXY_HTTP.equals(proxyInfo.getType())) {
+            routePlanner = new DefaultProxyRoutePlanner(new HttpHost(proxyInfo.getHost(), proxyInfo.getPort()));
+            if (proxyInfo.getUserName() != null) {
+                final Credentials creds;
+                if (proxyInfo.getNtlmHost() != null || proxyInfo.getNtlmDomain() != null) {
+                    creds = new NTCredentials(proxyInfo.getUserName(),
+                            proxyInfo.getPassword(),
+                            proxyInfo.getNtlmHost(),
+                            proxyInfo.getNtlmDomain());
+                } else {
+                    creds = new UsernamePasswordCredentials(proxyInfo.getUserName(),
+                            proxyInfo.getPassword());
+                }
+                AuthScope authScope = new AuthScope(proxyInfo.getHost(), proxyInfo.getPort());
+                credentialsProvider.setCredentials(authScope, creds);
+            }
+        } else {
+            routePlanner = new SystemDefaultRoutePlanner(ProxySelector.getDefault());
         }
+
+        final CloseableHttpClient httpClient = HttpClientBuilder.create()
+                .setConnectionManager(CONN_POOL)
+                .setConnectionManagerShared(true)
+                .setRoutePlanner(routePlanner)
+                .build();
+
+        final HttpFileRequester fileRequester = new HttpFileRequester(
+                httpClient,
+                this.session.getSettings().isInteractiveMode() ? getLog() : null);
```

Comment:
@ok2c instead of passing null and then check for null everywhere we introduce `SilentLog` class implementing `org.apache.maven.plugin.logging.Log`, which will simply do nothing.

This of course would imply that byte calculations are thrown away for non-interactive mode but let's avoid premature optimizations - readability is our first goal :)